cleanup.js

```diff
@@ -5,69 +5,119 @@
+const subscriptionIdEnvName = 'AZURE_SUBSCRIPTION_ID';
+const armEndpointEnvName = 'AZURE_ARM_ENDPOINT';
 
 _validateEnvironmentVariables();
 _validateParameters();
-var clientId = process.env['CLIENT_ID'];
-var domain = process.env['DOMAIN'];
-var secret = process.env['APPLICATION_SECRET'];
-var subscriptionId = process.env['AZURE_SUBSCRIPTION_ID'];
+
+var clientId = process.env[clientIdEnvName];
+var tenantId = process.env[tenantIdEnvName];
+var secret = process.env[secretEnvName];
+var subscriptionId = process.env[subscriptionIdEnvName];
+var armEndpoint = process.env[armEndpointEnvName];
 var resourceGroupName = process.argv[2];
-var resourceName = process.argv[3];
 var resourceClient;
-var resourceProviderNamespace = 'Microsoft.KeyVault';
-var parentResourcePath = '';
-var resourceType = 'vaults';
-var apiVersion = '2015-06-01';
+var map = {};
 
-function deleteResource(callback) {
-  console.log(util.format('\nDeleting resource:  %s'), resourceName);
-  return resourceClient.resources.deleteMethod(resourceGroupName, 
-                                               resourceProviderNamespace, 
-                                               parentResourcePath, 
-                                               resourceType, 
-                                               resourceName, 
-                                               apiVersion, 
-                                               callback);
-}
+const fetchUrl = armEndpoint + 'metadata/endpoints?api-version=1.0'
 
 function deleteResourceGroup(callback) {
-  console.log('\nDeleting resource group: ' + resourceGroupName);
+  console.log('\nStarting to delete resource group: ' + resourceGroupName);
   return resourceClient.resourceGroups.deleteMethod(resourceGroupName, callback);
 }
 
-
 function _validateEnvironmentVariables() {
   var envs = [];
-  if (!process.env['CLIENT_ID']) envs.push('CLIENT_ID');
-  if (!process.env['DOMAIN']) envs.push('DOMAIN');
-  if (!process.env['APPLICATION_SECRET']) envs.push('APPLICATION_SECRET');
-  if (!process.env['AZURE_SUBSCRIPTION_ID']) envs.push('AZURE_SUBSCRIPTION_ID');
+  if (!process.env[clientIdEnvName]) envs.push(clientIdEnvName);
+  if (!process.env[tenantIdEnvName]) envs.push(tenantIdEnvName);
+  if (!process.env[secretEnvName]) envs.push(secretEnvName);
+  if (!process.env[subscriptionIdEnvName]) envs.push(subscriptionIdEnvName);
   if (envs.length > 0) {
     throw new Error(util.format('please set/export the following environment variables: %s', envs.toString()));
   }
 }
 
 function _validateParameters() {
-  if (!process.argv[2] || !process.argv[3]) {
-    throw new Error('Please provide the resource group and the resource name by executing the script as follows: "node cleanup.js <resourceGroupName> <resourceName>".');
+  if (!process.argv[2]) {
+    throw new Error('Please provide the resource group and the resource name by executing the script as follows: "node cleanup.js <resourceGroupName>".');
```

Comment:
update error message
'Please provide the resource group name by executing the script as follows: "node cleanup.js <resourceGroupName>".'